### PR TITLE
Suppot For subscriptions using paypal 

### PR DESCRIPTION
**Pull request by martinbedouret**

### CODE REVIEW

File: api/controllers/subscription.js [hunk 1]
```diff
@@ -199,27 +208,32 @@ async function syncSubscriptions(req, res) {
   }
 }
 
-function mapRemoteSubscr(subscription) {
+function mapRemoteSubscr(subscription, paypalPlans) {
   const subscr = {
     subscriptionId: subscription.productId,
     name: subscription.listings[0].title,
     status: 'active',
     platform: 'android-playstore',
```
Comment (tomivm) [inline api/controllers/subscription.js:216]: now the platform is also Web

File: api/helpers/paypal.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+const BASE_URL = 'https://api-m.sandbox.paypal.com/v1/';
+
+const getAccessToken = async () => {
+    const headers = {
+        'Content-Type': 'application/x-www-form-urlencoded'
+    };
+    const auth = {
+        username: 'AZ2vK0luRWMX9zzwLs-Ko_B_TJxeHYvIFCgXWcNBt50wmj7oZcUw8n4cf11GgdClTVnYMuEs5vRnxVEk',
+        password: 'ENsTK3mMS1OEHi-HP-Lzwu4QYuiiv5no4f6jUFElHAhk3TwLDc1NJXBPrMwVs0K2jyyXpx3cCWbGrUni'
```
Comment (tomivm) [inline api/helpers/paypal.js:11]: password should be stored on an env variable?

**Review by tomivm — CHANGES_REQUESTED** [review]
:rocket:

File: api/controllers/subscriber.js [hunk 3]
```diff
@@ -108,6 +127,60 @@ async function getSubscriber(req, res) {
...
+        nativePurchase = remoteData;
+      } catch (err) {
+        console.log(err.message);
+      }
+      if (status) {
+        subscriber.status = status;
+        subscriber.transaction.expiryDate = expiryDate;
+        subscriber.transaction.nativePurchase = nativePurchase;
+        if (remoteData)
+          subscriber.save(function (err, subscr) {
```
Comment (RodriSanchez1) [inline api/controllers/subscriber.js:151]: Needs to await the save function. If not can return the 200 state on line 184 and fails after save. This breaks the server. Same on line 97. (Both lines on PR file)

File: api/controllers/subscription.js (hunk 1 above)
Comment (martinbedouret) [reply api/controllers/subscription.js:216]: The source of information to syncronize the subscription and plans is always the play store, then we link those subscriptions to the other platforms like Paypal

File: api/models/Subscribers.js [hunk 4]
```diff
@@ -74,7 +74,7 @@ subscribersSchema.pre('save', async function () {
   if (
     this.transaction?.state === 'approved'
   ) {
-    if (this.transaction.nativePurchase?.productId === this.product.subscriptionId) {
+    if (this.product.subscriptionId) {
       return true;
     }
     throw {
```
[the remark this answers is not in the document]
Comment (martinbedouret) [reply api/models/Subscribers.js:80]: fixed

File: api/controllers/subscriber.js (hunk 3 above)
Comment (martinbedouret) [reply api/controllers/subscriber.js:151]: fixed

**Pull request merged**